pkg/loop/internal/test/generic.go

```diff
@@ -0,0 +1,239 @@
+package test
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"reflect"
+
+	"google.golang.org/grpc"
+
+	"github.com/smartcontractkit/chainlink-relay/pkg/loop/internal"
+	"github.com/smartcontractkit/chainlink-relay/pkg/types"
+)
+
+type MockConn struct {
+	grpc.ClientConnInterface
+}
+
+type StaticGenericPluginMedianProvider struct {
```

Comment:
Generic was meant to be a (admittedly awkward) standin for "lightweight OCR". I've renamed this to StaticReportingPluginWithMedianProvider since Jordan recommended using ReportingPlugin throughout, and StaticGenericPluginProvider has been renamed to StaticReportingPluginWithPluginProvider.

Let me know if that makes sense :)